### ISSUE TITLE
gccgo: can shift by complex number

### ISSUE THREAD

**Issue by dvyukov** [golang/go#11574]
gccgo successfully compiles the following program:
```go
package a
var a = 0 << complex64(1)
```

go/types say:
```
go.go:2: invalid operation: 0 << complex64(1) (shift count type complex64, must be unsigned integer)
```

gccgo must unlearn how to shift by complex numbers.

https://github.com/golang/go/issues/11567

**Comment by dvyukov** [comment]
@paranoiacblack

**Comment by paranoiacblack** [comment]
I should note whatever changes fixes #11616, should also fix this.

**Comment by paranoiacblack** [comment]
#11616 is fixed.
